### PR TITLE
Next and Previous Pagination Component

### LINKED ISSUE #337: @/charterafrica/components/NextPrevPagination
Using MUI `Pagination`, implement `NexPreviousPagination` component per designs.

### Mobile

Image

### Desktop

Image

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

This PR implements the Next and Previous Pagination Components.

Fixes #337 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
nav

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
QQ @kelvinkipruto: Isn't this functionality implemented in @codeforafrica app? Any reason why we're not just reusing it with styling updates only?

File: apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { Typography, PaginationItem } from "@mui/material";
+import Box from "@mui/material/Box";
+import Pagination from "@mui/material/Pagination";
+import { styled } from "@mui/material/styles";
+import React from "react";
+
+import { neutral } from "@/charterafrica/colors";
+
+const StyledPagination = styled(Pagination)(() => ({
+  variant: "outlined",
```
Comment (koechkevin) [inline apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js:10]: @kelvinkipruto if not using theme then id doesn't have to be a function.

File: apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    borderRadius: "4px",
+    height: "33px",
+    border: "none",
+  },
+  "& .MuiPagination-ul": {
+    justifyContent: "center",
+  },
+}));
+
+function NavigationLabel(color, variant, text) {
```
Comment (koechkevin) [inline apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js:23]: Why do you have to redefine Typography as a function.

File: apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            hideDisabledButtons
+          ) {
+            return null;
+          }
+          return (
+            <PaginationItem
+              {...item}
+              slots={{
+                next: () =>
+                  NavigationLabel("neutral.dark", "p3SemiBold", "Next"),
```
Comment (koechkevin) [inline apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js:72]: Any reason why this should be a function call? `() => <Typography {...props}/>` should work

File: apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                previous: () =>
+                  NavigationLabel("neutral.light", "p3SemiBold", "Prev"),
```
Comment (koechkevin) [inline apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js:74]: ditto

File: apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js (hunk 3 above)
Comment (kelvinkipruto) [reply apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js:72]: @koechkevin This works but fails during linting. The main issues is [this](https://reactjs.org/docs/reconciliation.html#elements-of-different-types)

**Comment by kilemensi** [comment]
> QQ @kelvinkipruto: Isn't this functionality implemented in @CodeForAfrica app? Any reason why we're not just reusing it with styling updates only?

I asked ☝🏽 yesterday @kelvinkipruto and I'm asking again: Isn't this component exactly the same as https://github.com/CodeForAfrica/ui/tree/main/apps/codeforafrica/src/components/NextPreviousPagination (bar styling of course)?

**Comment by kelvinkipruto** [comment]
> `main`/apps/codeforafrica/src/components/NextPreviousPagination

@kilemensi Yes. I'm currently re-using it

**Comment by kilemensi** [comment]
> > `main`/apps/codeforafrica/src/components/NextPreviousPagination
> 
> @kilemensi Yes. I'm currently re-using it

How come [`NextPrev`](https://github.com/CodeForAfrica/ui/blob/ed64bdcd3af81137802464e70f961213d13fae06/apps/charterafrica/src/components/NextPrevPagination/NextPrevPagination.js) in this PR is very different from [`NextPrev`](https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/src/components/NextPreviousPagination/NextPreviousPagination.js) in codeforafrica app @kelvinkipruto ?

**Comment by kilemensi** [comment]
Any progress @kelvinkipruto ?

**Comment by kelvinkipruto** [comment]
@kilemensi @koechkevin I have updated this. Could you please leave your reviews?

**Review by koechkevin — APPROVED** [review]

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**